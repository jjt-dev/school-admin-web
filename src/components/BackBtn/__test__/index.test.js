import React from 'react'
import renderer from 'react-test-renderer'
import BackBtn from '..'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

const mockHistoryPush = jest.fn()

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

it('renders correctly back button', () => {
  const tree = renderer.create(<BackBtn />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('test the correct text is rendered', () => {
  render(<BackBtn />)
  // .toBeInTheDocument() is an assertion that comes from jest-dom
  // otherwise you could use .toBeDefined()
  expect(screen.getByText('返 回')).toBeInTheDocument()
})
