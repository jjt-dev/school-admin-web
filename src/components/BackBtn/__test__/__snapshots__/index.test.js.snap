// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly back button 1`] = `
<button
  className="ant-btn back-btn ant-btn-primary"
  onClick={[Function]}
  type="button"
>
  <span>
    返 回
  </span>
</button>
`;
